RecoParticleFlow/PFSimProducer/plugins/SimPFProducer.cc

```diff
@@ -84,8 +72,40 @@ class SimPFProducer : public edm::global::EDProducer<> {
   const std::vector<edm::EDGetTokenT<reco::TrackToTrackingParticleAssociator>> associators_;
 };
 
+#include "FWCore/Framework/interface/MakerMacros.h"
 DEFINE_FWK_MODULE(SimPFProducer);
 
+void SimPFProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // simPFProducer
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("simClustersSrc", {"particleFlowClusterHGCal"});
+  desc.add<edm::InputTag>("trackSrc", {"generalTracks"});
+  desc.add<std::vector<edm::InputTag>>("associators",
+                                       {
+                                           {"quickTrackAssociatorByHits"},
+                                       });
+  desc.add<edm::InputTag>("pfRecTrackSrc", {"hgcalTrackCollection", "TracksInHGCal"});
+  desc.add<edm::InputTag>("trackingParticleSrc", {"mix", "MergedTrackTruth"});
+  desc.add<double>("neutralEMThreshold", 0.25);
+  desc.add<edm::InputTag>("caloParticlesSrc", {"mix", "MergedCaloTruth"});
+  desc.add<double>("superClusterThreshold", 4.0);
+  desc.add<edm::InputTag>("simClusterTruthSrc", {"mix", "MergedCaloTruth"});
+  desc.add<edm::InputTag>("muonSrc", {"muons1stStep"});
+  desc.add<double>("neutralHADThreshold", 0.25);
+  desc.add<edm::InputTag>("gsfTrackSrc", {"electronGsfTracks"});
+
+  // if useTiming_
+  desc.add<edm::InputTag>("trackTimeValueMap", {});
```

Comment:
The logic of the code as it is now assumes that the bool `useTiming_` is `true` if `conf.existsAs<edm::InputTag>("trackTimeValueMap")`, and `false` otherwise.

But if you define "trackTimeValueMap" in the fillDescriptions method, then `useTiming_` can never be `false`.

This doesn't get exposed now because this simPFProducer module is only used in Phase2 with Timing, and therefore there is no current use case for `useTiming_` being `false`. However, this is a bug that must be fixed.

There are probably several ways out, but I think that the easiest and cleanest way for doing it is simply `addOptional` all these paramaters, without providing here any initialization value. As such those parameters will only get inserted by the modifier in RecoParticleFlow/Configuration/python/RecoParticleFlow_cff.py, and will not be present in unmodified configurations.

Reply:
Thanks a lot, I didn't know about `addOptional`!